src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2624,7 +2624,7 @@ private void EmitDefaultValue(TypeSymbol type, bool used, SyntaxNode syntaxNode)
             if (used)
             {
                 // default type parameter values must be emitted as 'initobj' regardless of constraints
-                if (!type.IsTypeParameter())
+                if (!type.IsTypeParameter() && type.SpecialType != SpecialType.System_Decimal)
```

Comment:
I didn't understand what is uniquely special about `Decimal`. Doesn't it have a default value? If yes, is it the only special type without a default value?